server/src/main/java/org/candlepin/dto/api/v1/ConsumerDTO.java

```diff
@@ -0,0 +1,985 @@
+        return recipientOwnerKey;
+    }
+
+    /**
+     * Sets the recipient owner key to set on this ConsumerDTO object.
+     *
+     * @param recipientOwnerKey the recipient owner key to set on this ConsumerDTO object.
+     *
+     * @return a reference to this DTO object.
+     */
+    public ConsumerDTO setRecipientOwnerKey(String recipientOwnerKey) {
+        this.recipientOwnerKey = recipientOwnerKey;
+        return this;
+    }
+
+    /**
+     * Retrieves the annotations field of this ConsumerDTO object.
+     *
+     * @return a reference to this DTO object.
+     */
+    public String getAnnotations() {
+        return annotations;
+    }
+
+    /**
+     * Sets the annotations to set on this ConsumerDTO object.
+     *
+     * @param annotations the annotations to set on this ConsumerDTO object.
+     *
+     * @return a reference to this DTO object.
+     */
+    public ConsumerDTO setAnnotations(String annotations) {
+        this.annotations = annotations;
+        return this;
+    }
+
+    /**
+     * Retrieves the content access mode field of this ConsumerDTO object.
+     *
+     * @return a reference to this DTO object.
+     */
+    public String getContentAccessMode() {
+        return this.contentAccessMode;
+    }
+
+    /**
+     * Sets the content access mode to set on this ConsumerDTO object.
+     *
+     * @param contentAccessMode the content access mode to set on this ConsumerDTO object.
+     *
+     * @return a reference to this DTO object.
+     */
+    public ConsumerDTO setContentAccessMode(String contentAccessMode) {
+        this.contentAccessMode = contentAccessMode;
+        return this;
+    }
+
+    /**
+     * Retrieves the type field of this ConsumerDTO object.
+     *
+     * @return a reference to this DTO object.
+     */
+    public ConsumerTypeDTO getType() {
+        return this.type;
+    }
+
+    /**
+     * Sets the consumer type to set on this ConsumerDTO object.
+     *
+     * @param type the type to set on this ConsumerDTO object.
+     *
+     * @return a reference to this DTO object.
+     */
+    public ConsumerDTO setType(ConsumerTypeDTO type) {
+        this.type = type;
+        return this;
+    }
+
+    /**
+     * Retrieves the id cert field of this ConsumerDTO object.
+     *
+     * @return a reference to this DTO object.
+     */
+    @HateoasArrayExclude
+    public CertificateDTO getIdCert() {
+        return this.idCert;
+    }
+
+    /**
+     * Sets the idCert to set on this ConsumerDTO object.
+     *
+     * @param idCert the idCert to set on this ConsumerDTO object.
+     *
+     * @return a reference to this DTO object.
+     */
+    public ConsumerDTO setIdCert(CertificateDTO idCert) {
+        this.idCert = idCert;
+        return this;
+    }
+
+    /**
+     * This will put in the property so that the virtWho instances won't error
+     *
+     * @return List always empty
+     */
+    @JsonProperty("guestIds")
+    public List<String> getEmptyGuestIdArray() {
```

Comment:
per the comment on the method in the existing model code, the need is virt-who clients.
This works for me, but please share alternative suggestions.

Reply:
What I mean is that if we need to always return a guest ID collection, maybe we need to update the DTO accordingly. This may be a good candidate for discussion when @cnsnyder returns so we can determine exactly what is needed, why it's needed and the cleanest way forward.

Reply:
* Based on the comment in the code, my understanding is that the absence of this collection would break (at least) older version of virt-who clients. Also we should ensure the dto work does not change the output of the APIs as they are today.
* we do need a getter that returns the actual guest ids, at the very least for test purposes, also to populate the entity itself.
* I like the idea to skip translation. translation seems like a waste anyway, so I changed the implementation to not translate guestIds.